openupgrade_scripts/scripts/hr_holidays/18.0.1.6/pre-migration.py

```diff
@@ -106,7 +125,7 @@ def split_employee_leaves(env):
             SELECT leave.id, 'company', array_agg(he.id) as employee_ids
             FROM {table} AS leave
             JOIN hr_employee he ON he.company_id = leave.mode_company_id
-            WHERE leave.holiday_type = 'company'
+            WHERE leave.holiday_type = 'company' and he.active = True
```

Comment:
this should be unnecessary, because if active != true, then by `update_states()` the state is cancel/refuse and thus it is already deleted by your first query in `split_employee_leaves()`

Reply:
Hère I am checking whether employee is active (in order to avoid creating leave/allocation for all past employees), the update_states is updating inactive leave/allocation, not employees

Reply:
Good point